### PR TITLE
Fix for Github Issue #37

### PR DESCRIPTION
**Pull request by kameshsampath**
Updated the XtextGenerate.xtend and BuildingASimpleXtendProject test case

### CODE REVIEW

**Comment by oehme** [comment]
Thank you for the PR. This fix won't work if you have Java files in packages (it currently only checks the top-level output folders). You can use Guava's Files#fileTreeTraverser. Please also adjust the test case to account for that.

**Comment by kameshsampath** [comment]
Alright, let me fix that up. I have already added the test case for the BuildingASimpleXtendProject#theGeneratorShouldRunAndCompileWhenInvokedInSeperateBuilds to make sure the fix works, do you need a  test case to filter the files ?

File: xtext-gradle-plugin/src/main/java/org/xtext/gradle/tasks/XtextGenerate.xtend [hunk 1]
```diff
@@ -151,6 +151,14 @@ class XtextGenerate extends DefaultTask {
 			initializeBuilder
 		}
 		val request = new GradleInstallDebugInfoRequest => [
+			//Issue #37 Fix
```
Comment (oehme) [inline xtext-gradle-plugin/src/main/java/org/xtext/gradle/tasks/XtextGenerate.xtend]: this comment is not needed, just include the issue number in your commit message

File: xtext-gradle-plugin/src/main/java/org/xtext/gradle/tasks/XtextGenerate.xtend [hunk 2]
```diff
@@ -151,6 +151,14 @@ class XtextGenerate extends DefaultTask {
...
+			if(generatedFiles.isNullOrEmpty){
+				val javaFiles = <File>newLinkedList
```
Comment (oehme) [inline xtext-gradle-plugin/src/main/java/org/xtext/gradle/tasks/XtextGenerate.xtend]: this additional local variable is not needed

**Comment by kameshsampath** [comment]
I have updated for the review comments and added the additional test case to cover the sources in Java package.

**Comment by oehme** [comment]
Awesome, thank you! =)

**Pull request merged**